test/e2e/multicast_test.go

```diff
@@ -158,7 +278,261 @@ type multicastTestcase struct {
+					From: []crdv1alpha1.NetworkPolicyPeer{},
+					Protocols: []crdv1alpha1.NetworkPolicyProtocol{
+						{
+							IGMP: &crdv1alpha1.IGMPProtocol{IGMPType: &igmpQueryType, GroupAddress: "224.0.0.1"},
+						},
+					},
+					Action: &anp.ruleConfigs[i].action,
+					Name:   anp.ruleConfigs[i].name,
+				})
+			} else {
+				np.Spec.Egress = append(np.Spec.Egress, crdv1alpha1.Rule{
+					From: []crdv1alpha1.NetworkPolicyPeer{},
+					Protocols: []crdv1alpha1.NetworkPolicyProtocol{{
+						IGMP: &crdv1alpha1.IGMPProtocol{GroupAddress: anp.ruleConfigs[i].address},
+					}},
+					Action: &anp.ruleConfigs[i].action,
+					Name:   anp.ruleConfigs[i].name,
+				})
+			}
+		}
+		if _, err = k8sUtils.CreateOrUpdateANP(np); err != nil {
+			failOnError(fmt.Errorf("create ANP failed for ANP %s: %v", np.Name, err), t)
+		}
+		failOnError(data.waitForANPRealized(t, data.testNamespace, np.Name), t)
+	}
+
+	for _, receiverConfig := range mc.receiverConfigs {
+		for _, addr := range receiverConfig.IPs {
+			go func(receiver, addr string) {
+				cmd := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -o -p %d -W %d %s", 2234, mcjoinWaitTimeout, addr)}
+				data.RunCommandFromPod(data.testNamespace, receiver, mcjoinContainerName, cmd)
+			}(receiverConfig.name, addr)
+		}
+	}
+
+	time.Sleep(5 * time.Second)
+
+	sendSessions := 10
+	var wg sync.WaitGroup
+	for i := 0; i < sendSessions; i++ {
+		wg.Add(1)
+		go func() {
+			for _, senderConfig := range mc.senderConfigs {
+				for _, addr := range senderConfig.IPs {
+					go func(sender, addr string) {
+						cmd := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 30 -c 1 -W %d %s", 2234, mcjoinWaitTimeout, addr)}
+						data.RunCommandFromPod(data.testNamespace, sender, mcjoinContainerName, cmd)
+					}(senderConfig.name, addr)
+				}
+			}
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	if err := wait.Poll(5*time.Second, defaultTimeout, func() (bool, error) {
+		for _, senderConfig := range mc.senderConfigs {
+			stats := mc.antctlResults[senderConfig.name]
+			t.Logf("check antctl get podmulticaststats result for %s", senderConfig.name)
+			antreaPod, err := data.getAntreaPodOnNode(senderConfig.nodeName)
+			if err != nil {
+				t.Errorf("error get getAntreaPodOnNode for %s", senderConfig.name)
+				failOnError(err, t)
+			}
+			isMatch, err := checkAntctlResult(t, data, antreaPod, senderConfig.name, stats.Inbound, stats.Outbound)
+			if err != nil || !isMatch {
+				return false, err
+			}
+		}
+		receiversAddresses := sets.NewString()
+		for _, receiverConfig := range mc.receiverConfigs {
+			receiversAddresses.Insert(receiverConfig.IPs...)
+			stats := mc.antctlResults[receiverConfig.name]
+			t.Logf("check antctl get podmulticaststats result for %s", receiverConfig.name)
+			antreaPod, err := data.getAntreaPodOnNode(receiverConfig.nodeName)
+			if err != nil {
+				t.Errorf("error get getAntreaPodOnNode for %s", receiverConfig.name)
+				failOnError(err, t)
```

Comment:
I didn't get your point with this `failOnError`. You didn't introduce new namespaces, what would you plan to cleanup? If it is targeting at clearing the ANP, you can use defer function after the ANP creation?

Reply:
Updated. Removed all the `failOnError` call in multicast e2e test file